sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -315,7 +322,80 @@ public override long Position
 
         public override long Seek(long offset, SeekOrigin origin)
         {
-            throw new NotSupportedException();
+            long newPosition = CalculateNewPosition(offset, origin);
+
+            if (newPosition == _position)
+            {
+                return _position;
+            }
+
+            // newPosition < 0
+            if (newPosition < 0)
+            {
+                throw new ArgumentException($"New {nameof(offset)} cannot be less than 0.  Value was {newPosition}");
+            }
+
+            // newPosition > _length
+            if (newPosition > _length)
+            {
+                if (_allowBlobModifications)
+                {
+                    _position = newPosition;
+                    _seeked = true;
+                    return newPosition;
+                }
```

Comment:
Should we check if length has been updated and validate it? What's the scenario where user would like to seek beyond length ?

Reply:
See my comment below, I'm not sure we can do network calls within Seek().